### ISSUE TITLE
Android: Heading property

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30877]
## Description
- No way to set something as an accessibility heading. 
- "Header" trait does not map to this as expected.

Missing feature that is supported on iOS but not Android.

## React Native version:
v0.63

**Comment by blavalla** [comment]
I can't reproduce this issue, it seems to be functioning as expected.

Snack - https://snack.expo.io/7tjnjlJbO

Closing out.

**Comment by blavalla** [comment]
Ah, I just realized what my original notes meant here (my bad on making them so vague).

While this does work as an "accessibilityRole", technically speaking on Android you can set something to have a role *and* be a heading, which you cannot do in React Native. I don't think this is a very common pattern to be honest, as none of the existing roles really lend themselves well to also being headings, and it feels more like a quirk of the API from Google's side than anything.

Technically though, if we wanted to support this, we'd need to split "header" out from accessibilityRole and create a new prop like accessibilityHeader={true} which maps identically to the way the header role maps now.

@kacieb, @lunaleaps, @nadiia, If you think this is worth re-opening, let me know and I can put the details in the description.
